pkg/start/start.go

```diff
@@ -48,9 +48,9 @@ const (
 	//leaseDuration = 90 * time.Second
 	//renewDeadline = 45 * time.Second
 	//retryPeriod   = 30 * time.Second
-	leaseDuration = 30 * time.Second
-	renewDeadline = 15 * time.Second
-	retryPeriod   = 10 * time.Second
+	leaseDuration = 137 * time.Second
```

Comment:
We should set these based on whether we're running in a SNO cluster, or even skip the leaderelection altogether in a SNO cluster since it's not HA, as to not lengthen shutdown/transition in an HA cluster.

Reply:
@jottofar For SNO even for non SNO situation the agreement is that each operator of OpenShift should withstand  tolerate 60s worth of disruption (e.g.  kube-apiserver is not available for up to 60 seconds) 

From https://github.com/openshift/enhancements/blob/master/CONVENTIONS.md
```
Leader election needs to be able to tolerate 60s worth of disruption. The default in library-go has been upgaded to handle this case in 4.9: https://github.com/openshift/library-go/blob/4b9033d00d37b88393f837a88ff541a56fd13621/pkg/config/leaderelection/leaderelection.go#L84 In essence, the kube-apiserver downtime tolerance is floor(renewDeadline/retryPeriod)*retryPeriod-retryPeriod. Recommended defaults are LeaseDuration=137s, RenewDealine=107s, RetryPeriod=26s. These are the configurable values in k8s.io/client-go based leases and controller-runtime exposes them. This gives us

    clock skew tolerance == 30s
    kube-apiserver downtime tolerance == 78s
    worst non-graceful lease reacquisition == 163s
    worst graceful lease reacquisition == 26s
```

Reply:
I am not sure if I understand clearly the side effects of  increasing the leaseDuration. AFAIU increase in the values will lead to situation where moving to a different version of CVO will more time. But that should not have a serious impact on the upgrade time in general.

Reply:
The bug reads as if this is a SNO driven change but looking at the references more closely, and IIUC, these changes are really 4.9 wide, not specific to SNO.

Reply:
Yeah, but I am also trying to figure out if these values are right for CVO or not. So trying to understand more.
```
// LeaseDuration is the duration that non-leader candidates will
	// wait to force acquire leadership. This is measured against time of
	// last observed ack.
	LeaseDuration time.Duration
	// RenewDeadline is the duration that the acting master will retry
	// refreshing leadership before giving up.
	RenewDeadline time.Duration
	// RetryPeriod is the duration the LeaderElector clients should wait
	// between tries of actions.
	RetryPeriod time.Duration
```
From https://raw.githubusercontent.com/kubernetes/contrib/master/election/vendor/k8s.io/kubernetes/pkg/client/leaderelection/leaderelection.go

Reply:
Raising `renewDeadline` sounds useful for the single-node case, unless the CVO would fail out based on some other caller failing to find the API-server.

Fiddling with `retryPeriod` doesn't seem particularly relevant, so I'm in favor of just picking up the library-go default to reduce the number of special snowflakes.

Raising `leaseDuration` will make us slower to recover from CVO's dying without gracefully releasing our leases.  But we're a lot better about graceful lease releases than we were back in #426, so this is another case where I don't think it matters all that much anymore, and I'm in favor of moving to the library-go defaults here too.

Reply:
Can we update the commit message with some of this motivation/backstory^^?

Reply:
Tried to fix it in https://github.com/openshift/cluster-version-operator/commit/d39b5c56b794e89254f7fd8771554d59e8785b26